src/matrices.rs

```diff
@@ -0,0 +1,580 @@
+#![allow(dead_code)]
+
+use crate::traits::MatrixNumber;
+use crate::traits::{CheckedMulScl, LaTeXable};
+use anyhow::Context;
+use num_traits::{CheckedAdd, CheckedMul, CheckedNeg, CheckedSub};
+use std::ops::{Add, Mul, Neg, Sub};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Matrix<T: MatrixNumber> {
+    data: Vec<Vec<T>>,
+}
+
+#[derive(Debug, Clone)]
+struct Aftermath<T: MatrixNumber> {
+    result: Matrix<T>,
+    steps: Vec<String>,
+}
+
+impl<T: MatrixNumber> Matrix<T> {
+    fn new(data: Vec<Vec<T>>) -> Self {
+        Self { data }
+    }
+
+    pub fn new_safe(data: Vec<Vec<T>>) -> Self {
```

Comment:
to nie jest safe, tylko tamta druga metoda jest unsafe. Zazwyczaj robi się jedną metodę zwracającą result i drugą panikującą (ew. trzecią robiącą UB: https://doc.rust-lang.org/std/primitive.slice.html#method.get_unchecked). Ale tutaj to chyba w ogóle nie ma sensu robić dwóch metod.